Composer/packages/lib/indexers/src/dialogUtils/dialogDiff.ts

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+/**
+ * not like JsonDiff, DialogDiff compaire stop at { $kind, $id, ... }
+ * The atomic unit to add/delete/update is a dialog object {}
+ */
+
+import has from 'lodash/has';
+import get from 'lodash/get';
+import isEqual from 'lodash/isEqual';
+import { SDKKinds } from '@bfc/shared';
+
+import {
+  JsonDiffAdds,
+  JsonDiffUpdates,
+  IJsonChanges,
+  IJSONChangeAdd,
+  IJSONChangeDelete,
+  IJSONChangeUpdate,
+  IComparator,
+  IStopper,
```

Comment:
In this case, Comparator is a generic thing, which make sense, but Stopper feels weird, stopper is just a sepecifc process in your walker, if i understand correctly.